packages/openneuro-app/src/scripts/dataset/mutations/dataset-events.tsx

```diff
@@ -78,12 +46,50 @@ export const DatasetEvents = ({ datasetId }) => {
       setUpdatedNote("")
       refetch()
     },
-    onError: (error) => {
-      Sentry.captureException(error)
+    onError: (mutationError) => {
+      Sentry.captureException(mutationError)
+      toast.error(
+        <ToastContent
+          title="Error updating note"
+          body={mutationError.message}
+        />,
+      )
     },
   })
 
-  const handleAddEvent = () => {
+  const rawEvents = data?.dataset?.events || []
+
+  const processedEvents = useMemo(() => {
+    const responsesMap = new Map()
+    rawEvents.forEach((event) => {
+      if (event.event.type === "contributorResponse" && event.event.requestId) {
+        responsesMap.set(event.event.requestId, event)
+      }
+    })
+
+    const enrichedEvents = rawEvents.map((event) => {
+      if (event.event.type === "contributorRequest") {
+        const response = responsesMap.get(event.id)
+        if (response) {
+          return {
+            ...event,
+            hasBeenRespondedTo: true,
+            responseStatus: response.event.status,
+          }
+        }
+      }
+      return event
+    })
+    return enrichedEvents
+  }, [rawEvents])
```

Comment:
This step could be done on the API side instead. If the client needs this, the API should just return it or provide an option for it.

Reply:
moved to the backend